stream/stream_slice.c

```diff
@@ -151,8 +151,9 @@ static int open2(struct stream *stream, struct stream_open_args *args)
         return parse_ret;
     }
 
-    args->url = stream->path;
-    int inner_ret = stream_create_with_args(args, &p->inner);
+    struct stream_open_args args2 = *args;
```

Comment:
Can you expand a bit on who frees what, and what gets used after free?

Reply:
the `stream` talloc context (and with it `stream->path`) gets freed when open2 returns an error in line 158.
Setting `args->url` overwrites the original open URL(\*) and tricks stream.c into reusing the freed string for the next [loop iteration](https://github.com/mpv-player/mpv/blob/c046fed186eb442caf241cf1d5f3ff693ddd864e/stream/stream.c#L426-L435).
(\*) doing that is an awful idea in general

Reply:
Ouch. That's seriously non obvious from reading the (old or new) code. This really needs a comment at the code.